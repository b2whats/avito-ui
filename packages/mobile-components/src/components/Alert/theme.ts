import { CardTheme, dsl } from '@avito/core'

export const alertTheme = dsl.theme<CardTheme>()
  .slot('Card', {
    props: {
      px: 16,
      pt: 10,
      pb: 12,
    },
  })
  .slot('Close', {
    props: {

    },
  })
  .slot('Image', {
    props: {
      size: 20,
      mr: 12,
    },
  })
  .slot('Content', {})
  .build()
