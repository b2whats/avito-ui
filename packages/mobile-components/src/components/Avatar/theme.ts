import { AvatarTheme, DeepPartial } from '@avito/core'

export const avatarTheme: DeepPartial<AvatarTheme> = {
  scheme: {
    Wrapper: {
      style: {
        bg: 'gray4',
      },
      type: {
        shop: {
          style: {
            borderRadius: 5,
          },
        },
      },
    },
  },
}
