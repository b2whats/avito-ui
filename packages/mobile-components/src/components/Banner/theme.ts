import { CardTheme, dsl } from '@avito/core'


export const bannerTheme = dsl.theme<CardTheme>()
  .slot('Image', {
    p: 0,
  })
  .slot('Content', slot => [
    {
      p: 16,
      pt: 10,
    },
    slot.if('image', {
      pr: 8,
    }),
    slot.if('column', {
      p: 10,
    }),
  ])
  .build()
