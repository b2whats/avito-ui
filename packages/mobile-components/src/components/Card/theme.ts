import { CardTheme, dsl } from '@avito/core'

export const cardTheme = dsl.theme<CardTheme>()
  .slot('Card', {
    props: {
      rounded: 5,
    },
  })
  .slot('Close', {
    props: {
      p: 4,
    },
  })
  .slot('Image')
  .slot('Content', {
    props: {
      p: 10,
    },
  })
  .build()
