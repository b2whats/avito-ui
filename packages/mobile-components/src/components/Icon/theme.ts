import { createComponentTheme } from '../../theme/'
import { SchemeType } from '../../styled-system/'
import { IconProps } from './contract'
import { IconProps as IconBaseProps } from '@avito/icons'

export type IconTheme = {
  scheme: {
    Icon: SchemeType<IconProps, IconBaseProps>,
  }
}

const iconTheme: IconTheme = {
  scheme: {
    Icon: {
      size: {
        s: {
          props: {
            size: 16,
          },
        },
        m: {
          props: {
            size: 20,
          },
        },
        l: {
          props: {
            size: 24,
          },
        },
      },
      variant: {
        primary: {
          style: {
            color: 'blue500',
          },
        },
        secondary: {
          style: {
            color: 'gray500',
          },
        },
        error: {
          style: {
            color: 'red500',
          },
        },
        success: {
          style: {
            color: 'green500',
          },
        },
        warning: {
          style: {
            color: 'yellow500',
          },
        },
      },
    },
  },
}

export const createIconTheme = createComponentTheme('Icon', iconTheme)