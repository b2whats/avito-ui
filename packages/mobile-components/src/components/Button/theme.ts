import { SchemeType } from '@avito/core/styled-system//'
import { ButtonProps } from './contract'
import { TextProps } from '../Text/contract'
import { IconProps } from '../Icon/contract'
import { SpinnerProps } from '../Spinner/contract'

export type ButtonTheme = {
  scheme: {
    Button: SchemeType<ButtonProps>,
    IconBefore: SchemeType<ButtonProps, IconProps>,
    Text: SchemeType<ButtonProps, TextProps>,
    IconAfter: SchemeType<ButtonProps, IconProps>,
    Spinner: SchemeType<ButtonProps, SpinnerProps>,
  },
}

export const buttonTheme: ButtonTheme = {
  scheme: {
    Spinner: {
      size: {
        s: {
          props: {
            size: 's',
          },
        },
        m: {
          props: {
            size: 'm',
          },
        },
        l: {
          props: {
            size: 'l',
          },
        },
      },
    },
    IconBefore: {
      size: {
        s: {
          props: {
            size: 's',
          },
        },
        m: {
          props: {
            size: 's',
          },
        },
        l: {
          props: {
            size: 'm',
          },
        },
      },
    },
    IconAfter: {
      size: {
        s: {
          props: {
            size: 's',
          },
        },
        m: {
          props: {
            size: 's',
          },
        },
        l: {
          props: {
            size: 'm',
          },
        },
      },
    },
    Text: {
      props: {
        size: 's',
      },
      iconBefore: {
        props: {
          ml: 6,
        },
      },
      iconAfter: {
        props: {
          mr: 6,
        },
      },
      kind: {
        flat: {
          props: {
            uppercase: true,
          },
        },
      },
    },
    Button: {
      style: {
        borderRadius: 3,
        px: 16,
        borderWidth: 1,
        borderStyle: 'solid',
        borderColor: 'transparent',
        focus: true,
      },
      size: {
        s: {
          style: {
            py: 4,
            minHeight: 's',
          },
        },
        m: {
          style: {
            py: 6,
            minHeight: 'm',
          },
        },
        l: {
          style: {
            py: 8,
            minHeight: 'l',
          },
        },
      },
      iconBefore: {
        children: {
          style: {
            pl: 12,
          },
        },
      },
      iconAfter: {
        children: {
          style: {
            pr: 12,
          },
        },
      },
      preset: {
        primary: {
          style: {
            color: 'white',
            colorDisabled: 'blue50',
            bg: 'blue500',
            bgActive: 'blue800',
            bgDisabled: 'blue300',
          },
        },
        accent: {
          style: {
            color: 'white',
            colorDisabled: 'gray48',
            bg: 'green500',
            bgActive: 'green800',
            bgDisabled: 'gray4',
          },
        },
        secondary: {
          style: {
            color: 'blue500',
            colorDisabled: 'gray48',
            bg: 'blue50',
            bgActive: 'blue100',
            bgDisabled: 'gray4',
          },
        },
        default: {
          style: {
            color: 'black',
            colorDisabled: 'gray48',
            bg: 'gray4',
            bgActive: 'gray8',
            bgDisabled: 'gray4',
          },
          checked: {
            style: {
              color: 'white',
              colorDisabled: 'blue50',
              bg: 'blue500',
              bgActive: 'blue800',
              bgDisabled: 'blue300',
            },
          },
        },
        defaultOnSurface: {
          style: {
            color: 'black',
            colorDisabled: 'gray48',
            bg: 'gray8',
            bgActive: 'gray12',
            bgDisabled: 'gray8',
          },
        },
        defaultDark: {
          style: {
            color: 'white',
            colorDisabled: 'gray40',
            bg: 'gray68',
            bgActive: 'gray76',
            bgDisabled: 'gray76',
          },
        },
        outline: {
          style: {
            color: 'black',
            colorDisabled: 'gray48',
            bg: 'white',
            borderColor: 'gray8',
            bgActive: 'gray8',
            //bgDisabled: 'gray4',
          },
        },
        pay: {
          style: {
            color: 'white',
            colorDisabled: 'gray40',
            bg: 'black',
            bgActive: 'gray76',
            bgDisabled: 'gray68',
          },
        },
        appInstall: {
          style: {
            color: 'blue500',
            colorDisabled: 'gray48',
            bg: 'gray4',
            bgActive: 'gray8',
            bgDisabled: 'gray4',
          },
        },
        linkIncreased: {
          style: {
            color: 'blue500',
            colorDisabled: 'gray40',
            bg: 'white',
            bgActive: 'gray4',
            bgDisabled: 'white',
          },
        },
      },
      kind: {
        default: {
          style: {
            bg: 'blue50',
            bgHover: 'blue60',
            bgActive: 'blue70',
            bgChecked: 'blue70',
            bgDisabled: 'blue30',
            color: 'white',
            colorDisabled: 'gray30',
          },
          variant: {
            primary: {
              style: {
                bg: 'blue50',
                bgHover: 'blue60',
                bgActive: 'blue70',
                bgChecked: 'blue70',
                bgDisabled: 'blue30',
              },
            },
            secondary: {
              style: {
                bg: 'gray50',
                bgHover: 'gray60',
                bgActive: 'gray70',
                bgChecked: 'gray70',
                bgDisabled: 'gray30',
                color: 'black',
                colorDisabled: 'gray70',
              },
            },
            success: {
              style: {
                bg: 'green50',
                bgHover: 'green60',
                bgActive: 'green70',
                bgChecked: 'green70',
                bgDisabled: 'green30',
              },
            },
            warning: {
              style: {
                bg: 'yellow50',
                bgHover: 'yellow60',
                bgActive: 'yellow70',
                bgChecked: 'yellow70',
                bgDisabled: 'yellow30',
                color: 'black',
                colorDisabled: 'gray70',
              },
            },
            error: {
              style: {
                bg: 'red50',
                bgHover: 'red60',
                bgActive: 'red70',
                bgChecked: 'red70',
                bgDisabled: 'red30',
              },
            },
          },
        },
        outline: {
          style: {
            borderColor: 'gray32',
            borderColorHover: 'gray40',
            borderColorActive: 'gra52',
            borderColorDisabled: 'gray24',
            bg: 'gray32',
            bgHover: 'gray40',
            bgActive: 'gray52',
            bgChecked: 'gray32',
          },
          variant: {
            primary: {
              style: {
                color: 'blue500',
                colorHover: 'blue600',
                colorActive: 'blue700',
                colorDisabled: 'blue300',
              },
              checked: {
                style: {
                  color: 'red500',
                  colorHover: 'red600',
                  colorActive: 'red700',
                  colorDisabled: 'red300',
                  bg: 'red300',
                },
              },
            },
            secondary: {
              style: {
                color: 'gray50',
                colorHover: 'gray60',
                colorActive: 'gray70',
                colorDisabled: 'gray30',
              },
            },
            success: {
              style: {
                color: 'green50',
                colorHover: 'green60',
                colorActive: 'green70',
                colorDisabled: 'green30',
              },
            },
            warning: {
              style: {
                color: 'yellow50',
                colorHover: 'yellow60',
                colorActive: 'yellow70',
                colorDisabled: 'yellow30',
              },
            },
            error: {
              style: {
                color: 'red50',
                colorHover: 'red60',
                colorActive: 'red70',
                colorDisabled: 'red30',
              },
            },
          },
        },
        flat: {
          style: {
            borderColor: 'gray30',
            borderColorHover: 'gray40',
            borderColorActive: 'gra50',
            borderColorDisabled: 'gray20',
            bg: 'gray30',
            bgHover: 'gray40',
            bgActive: 'gray50',
            bgChecked: 'gray30',
          },
          variant: {
            primary: {
              style: {
                color: 'blue50',
                colorHover: 'blue60',
                colorActive: 'blue70',
                colorDisabled: 'blue30',
              },
            },
            secondary: {
              style: {
                color: 'gray50',
                colorHover: 'gray60',
                colorActive: 'gray70',
                colorDisabled: 'gray30',
              },
            },
            success: {
              style: {
                color: 'green50',
                colorHover: 'green60',
                colorActive: 'green70',
                colorDisabled: 'green30',
              },
            },
            warning: {
              style: {
                color: 'yellow50',
                colorHover: 'yellow60',
                colorActive: 'yellow70',
                colorDisabled: 'yellow30',
              },
            },
            error: {
              style: {
                color: 'red50',
                colorHover: 'red60',
                colorActive: 'red70',
                colorDisabled: 'red30',
              },
            },
          },
        },
      },
    },
  },
}