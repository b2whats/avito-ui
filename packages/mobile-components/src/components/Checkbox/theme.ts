import { CheckboxTheme, dsl } from '@avito/core'
import { CheckboxCheckedIcon, CheckboxIndeterminateIcon } from '../Icon'

export const checkboxTheme = dsl.theme<CheckboxTheme>()
  .slot('Icon', slot => [
    slot.if('checked', {
      component: CheckboxCheckedIcon,
    }),
    slot.if('indeterminate', {
      component: CheckboxIndeterminateIcon,
    }),
  ])
  .slot('Switch', slot => [
    {
      style: {
        borderRadius: 2,
        height: 18,
        bg: 'gray8',
        bgActive: 'gray12',
        color: 'white',
        bgDisabled: 'gray4',
      },
    },
    slot.if('checked', [
      {
        style: {
          bgDisabled: 'gray28',
        },
      },
      slot.switch('variant', dsl.styleMap({
        primary: {
          bg: 'blue500',
          bgActive: 'blue700',
        },
        secondary: {
          bg: 'gray48',
          bgActive: 'gray56',
        },
        success: {
          bg: 'green500',
          bgActive: 'green700',
        },
        warning: {
          bg: 'orange500',
          bgActive: 'orange700',
        },
        error: {
          bg: 'red500',
          bgActive: 'red700',
        },
      })),
    ]),
  ])
  .build()
