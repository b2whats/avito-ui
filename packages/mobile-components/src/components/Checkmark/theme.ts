import { ComponentTheme, Slot, IconProps } from '@avito/core'
import { CheckmarkIcon } from '../Icon'
import { CheckmarkProps } from './contract'

export type CheckmarkTheme = ComponentTheme<CheckmarkProps, { Icon: Slot<IconProps> }>

export const checkmarkTheme: CheckmarkTheme = {
  defaultProps: {
    variant: 'primary',
  },
  scheme: {
    Icon: {
      component: CheckmarkIcon,
      disabled: {
        props: {
          color: 'gray28',
        },
      },
    },
  },
}
