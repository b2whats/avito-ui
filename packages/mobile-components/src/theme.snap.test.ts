import {
  mergeTheme,
  foldThemeParams,
  buttonTheme,
  inputTheme,
  textTheme,
  checkboxTheme,
  toggleTheme,
  textareaTheme,
  radioTheme,
  switcherTheme
} from '@avito/core'
import { theme } from './theme'

const variant = ['primary', 'secondary', 'success', 'error', 'warning']
const withToggle = (override: any) => mergeTheme(toggleTheme, theme.Toggle, override)

describe('mobile theme', () => {
  it('button theme snap', () => {
    expect(mergeTheme(buttonTheme, theme.Button)).toMatchSnapshot()
  })
  it('input theme snap', () => {
    expect(mergeTheme(inputTheme, theme.Input)).toMatchSnapshot()
  })
  it('text theme snap', () => {
    expect(mergeTheme(textTheme, theme.Text)).toMatchSnapshot()
  })

  describeTheme('checkbox', withToggle(mergeTheme(checkboxTheme, theme.Checkbox)), {
    variant,
    checked: [true, false],
    shape: ['circle', 'square'],
  })

  describeTheme('radio', withToggle(mergeTheme(radioTheme, theme.Radio)), {
    variant,
    checked: [true, false],
  })

  describeTheme('switcher', withToggle(mergeTheme(switcherTheme, theme.Switcher)), {
    variant,
    checked: [true, false],
    loading: [true, false],
  })

  describeTheme('textarea', mergeTheme(textareaTheme, theme.Textarea), {
    variant,
    clearable: [true, false],
    size: ['s', 'm', 'l'],
  })
})

function describeTheme(name: string, theme: any, propValues: { [K: string]: any[] }) {
  describe(name, () => {
    it('snap', () => expect(theme).toMatchSnapshot())
    it('fuzz', () => fuzz(propValues, theme))
  })
}

function cartesian<T extends { [K: string]: any[] }>(valuesByKey: T): { [K in keyof T]: any }[] {
  let res: any[] = [{}]
  Object.entries(valuesByKey).forEach(([key, values]) => {
    res = values.reduce(
      (acc, value) => [...acc, ...res.map(item => ({ ...item, [key]: value }))],
      [])
  })
  return res
}

function fuzz(propValues: { [K: string]: any[] }, theme: any) {
  cartesian(propValues).forEach(props => {
    expect(foldThemeParams(props, theme)).toMatchSnapshot(JSON.stringify(props))
  })
}
