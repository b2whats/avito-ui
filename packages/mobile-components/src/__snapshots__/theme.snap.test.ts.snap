// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mobile theme button theme snap 1`] = `
Object {
  "defaultProps": Object {
    "preset": "primary",
    "size": "m",
    "type": "button",
  },
  "scheme": Object {
    "Button": Object {
      "iconAfter": Object {
        "children": Object {
          "style": Object {
            "pr": 12,
          },
        },
      },
      "iconBefore": Object {
        "children": Object {
          "style": Object {
            "pl": 12,
          },
        },
      },
      "kind": Object {
        "default": Object {
          "style": Object {
            "bg": "blue50",
            "bgActive": "blue70",
            "bgChecked": "blue70",
            "bgDisabled": "blue30",
            "bgHover": "blue60",
            "color": "white",
            "colorDisabled": "gray30",
          },
          "variant": Object {
            "error": Object {
              "style": Object {
                "bg": "red50",
                "bgActive": "red70",
                "bgChecked": "red70",
                "bgDisabled": "red30",
                "bgHover": "red60",
              },
            },
            "primary": Object {
              "style": Object {
                "bg": "blue50",
                "bgActive": "blue70",
                "bgChecked": "blue70",
                "bgDisabled": "blue30",
                "bgHover": "blue60",
              },
            },
            "secondary": Object {
              "style": Object {
                "bg": "gray50",
                "bgActive": "gray70",
                "bgChecked": "gray70",
                "bgDisabled": "gray30",
                "bgHover": "gray60",
                "color": "black",
                "colorDisabled": "gray70",
              },
            },
            "success": Object {
              "style": Object {
                "bg": "green50",
                "bgActive": "green70",
                "bgChecked": "green70",
                "bgDisabled": "green30",
                "bgHover": "green60",
              },
            },
            "warning": Object {
              "style": Object {
                "bg": "yellow50",
                "bgActive": "yellow70",
                "bgChecked": "yellow70",
                "bgDisabled": "yellow30",
                "bgHover": "yellow60",
                "color": "black",
                "colorDisabled": "gray70",
              },
            },
          },
        },
        "flat": Object {
          "style": Object {
            "bg": "gray30",
            "bgActive": "gray50",
            "bgChecked": "gray30",
            "bgHover": "gray40",
            "borderColor": "gray30",
            "borderColorActive": "gra50",
            "borderColorDisabled": "gray20",
            "borderColorHover": "gray40",
          },
          "variant": Object {
            "error": Object {
              "style": Object {
                "color": "red50",
                "colorActive": "red70",
                "colorDisabled": "red30",
                "colorHover": "red60",
              },
            },
            "primary": Object {
              "style": Object {
                "color": "blue50",
                "colorActive": "blue70",
                "colorDisabled": "blue30",
                "colorHover": "blue60",
              },
            },
            "secondary": Object {
              "style": Object {
                "color": "gray50",
                "colorActive": "gray70",
                "colorDisabled": "gray30",
                "colorHover": "gray60",
              },
            },
            "success": Object {
              "style": Object {
                "color": "green50",
                "colorActive": "green70",
                "colorDisabled": "green30",
                "colorHover": "green60",
              },
            },
            "warning": Object {
              "style": Object {
                "color": "yellow50",
                "colorActive": "yellow70",
                "colorDisabled": "yellow30",
                "colorHover": "yellow60",
              },
            },
          },
        },
        "outline": Object {
          "style": Object {
            "bg": "gray32",
            "bgActive": "gray52",
            "bgChecked": "gray32",
            "bgHover": "gray40",
            "borderColor": "gray32",
            "borderColorActive": "gra52",
            "borderColorDisabled": "gray24",
            "borderColorHover": "gray40",
          },
          "variant": Object {
            "error": Object {
              "style": Object {
                "color": "red50",
                "colorActive": "red70",
                "colorDisabled": "red30",
                "colorHover": "red60",
              },
            },
            "primary": Object {
              "checked": Object {
                "style": Object {
                  "bg": "red300",
                  "color": "red500",
                  "colorActive": "red700",
                  "colorDisabled": "red300",
                  "colorHover": "red600",
                },
              },
              "style": Object {
                "color": "blue500",
                "colorActive": "blue700",
                "colorDisabled": "blue300",
                "colorHover": "blue600",
              },
            },
            "secondary": Object {
              "style": Object {
                "color": "gray50",
                "colorActive": "gray70",
                "colorDisabled": "gray30",
                "colorHover": "gray60",
              },
            },
            "success": Object {
              "style": Object {
                "color": "green50",
                "colorActive": "green70",
                "colorDisabled": "green30",
                "colorHover": "green60",
              },
            },
            "warning": Object {
              "style": Object {
                "color": "yellow50",
                "colorActive": "yellow70",
                "colorDisabled": "yellow30",
                "colorHover": "yellow60",
              },
            },
          },
        },
      },
      "preset": Object {
        "accent": Object {
          "style": Object {
            "bg": "green500",
            "bgActive": "green800",
            "bgDisabled": "gray4",
            "color": "white",
            "colorDisabled": "gray48",
          },
        },
        "appInstall": Object {
          "style": Object {
            "bg": "gray4",
            "bgActive": "gray8",
            "bgDisabled": "gray4",
            "color": "blue500",
            "colorDisabled": "gray48",
          },
        },
        "default": Object {
          "checked": Object {
            "style": Object {
              "bg": "blue500",
              "bgActive": "blue800",
              "bgDisabled": "blue300",
              "color": "white",
              "colorDisabled": "blue50",
            },
          },
          "style": Object {
            "bg": "gray4",
            "bgActive": "gray8",
            "bgDisabled": "gray4",
            "color": "black",
            "colorDisabled": "gray48",
          },
        },
        "defaultDark": Object {
          "style": Object {
            "bg": "gray68",
            "bgActive": "gray76",
            "bgDisabled": "gray76",
            "color": "white",
            "colorDisabled": "gray40",
          },
        },
        "defaultOnSurface": Object {
          "style": Object {
            "bg": "gray8",
            "bgActive": "gray12",
            "bgDisabled": "gray8",
            "color": "black",
            "colorDisabled": "gray48",
          },
        },
        "linkIncreased": Object {
          "style": Object {
            "bg": "white",
            "bgActive": "gray4",
            "bgDisabled": "white",
            "color": "blue500",
            "colorDisabled": "gray40",
          },
        },
        "outline": Object {
          "style": Object {
            "bg": "white",
            "bgActive": "gray8",
            "borderColor": "gray8",
            "color": "black",
            "colorDisabled": "gray48",
          },
        },
        "pay": Object {
          "style": Object {
            "bg": "black",
            "bgActive": "gray76",
            "bgDisabled": "gray68",
            "color": "white",
            "colorDisabled": "gray40",
          },
        },
        "primary": Object {
          "style": Object {
            "bg": "blue500",
            "bgActive": "blue600",
            "bgChecked": "blue600",
            "bgDisabled": "blue300",
            "color": "white",
            "colorDisabled": "blue50",
          },
        },
        "secondary": Object {
          "style": Object {
            "bg": "blue50",
            "bgActive": "blue100",
            "bgDisabled": "gray4",
            "color": "blue500",
            "colorDisabled": "gray48",
          },
        },
      },
      "size": Object {
        "l": Object {
          "style": Object {
            "minHeight": "l",
            "py": 8,
          },
        },
        "m": Object {
          "style": Object {
            "minHeight": "m",
            "py": 6,
          },
        },
        "s": Object {
          "style": Object {
            "minHeight": "s",
            "py": 4,
          },
        },
      },
      "style": Object {
        "borderColor": "transparent",
        "borderRadius": 3,
        "borderStyle": "solid",
        "borderWidth": 1,
        "focus": false,
        "px": 16,
      },
    },
    "IconAfter": Object {
      "size": Object {
        "l": Object {
          "props": Object {
            "size": "m",
          },
        },
        "m": Object {
          "props": Object {
            "size": "s",
          },
        },
        "s": Object {
          "props": Object {
            "size": "s",
          },
        },
      },
    },
    "IconBefore": Object {
      "size": Object {
        "l": Object {
          "props": Object {
            "size": "m",
          },
        },
        "m": Object {
          "props": Object {
            "size": "s",
          },
        },
        "s": Object {
          "props": Object {
            "size": "s",
          },
        },
      },
    },
    "Spinner": Object {
      "size": Object {
        "l": Object {
          "props": Object {
            "size": "l",
          },
        },
        "m": Object {
          "props": Object {
            "size": "m",
          },
        },
        "s": Object {
          "props": Object {
            "size": "s",
          },
        },
      },
    },
    "Text": Object {
      "iconAfter": Object {
        "props": Object {
          "mr": 6,
        },
      },
      "iconBefore": Object {
        "props": Object {
          "ml": 6,
        },
      },
      "kind": Object {
        "flat": Object {
          "props": Object {
            "uppercase": true,
          },
        },
      },
      "props": Object {
        "dense": true,
        "ml": -1,
        "size": "m",
      },
    },
  },
}
`;

exports[`mobile theme input theme snap 1`] = `
Object {
  "defaultProps": Object {
    "kind": "default",
    "size": "m",
    "variant": "primary",
  },
  "deletePlaceholderOnFocus": true,
  "scheme": Object {
    "IconAfter": Object {
      "clearable": Object {
        "component": [Function],
        "props": Object {
          "area": 10,
        },
      },
      "props": Object {
        "color": "gray28",
      },
      "size": Object {
        "l": Object {
          "props": Object {
            "mx": 10,
            "size": "m",
          },
        },
        "m": Object {
          "props": Object {
            "mx": 10,
            "size": "s",
          },
        },
        "s": Object {
          "props": Object {
            "mx": 8,
            "size": "s",
          },
        },
      },
    },
    "IconBefore": Object {
      "props": Object {
        "color": "gray28",
        "ml": 10,
        "mr": 8,
      },
      "size": Object {
        "l": Object {
          "props": Object {
            "size": "m",
          },
        },
        "m": Object {
          "props": Object {
            "size": "s",
          },
        },
        "s": Object {
          "props": Object {
            "size": "s",
          },
        },
      },
    },
    "Input": Object {
      "size": Object {
        "l": Object {
          "style": Object {
            "fontSize": 16,
            "height": "l",
          },
        },
        "m": Object {
          "style": Object {
            "fontSize": 16,
            "height": "m",
          },
        },
        "s": Object {
          "style": Object {
            "fontSize": 12,
            "height": "s",
          },
        },
      },
      "style": Object {
        "borderRadius": 5,
        "color": "black",
        "colorDisabled": "gray40",
        "focus": true,
        "placeholderColor": "gray48",
      },
      "variant": Object {
        "error": Object {
          "style": Object {
            "bg": "red50",
          },
        },
        "primary": Object {
          "style": Object {
            "bg": "gray4",
          },
        },
        "secondary": Object {
          "style": Object {
            "bg": "gray8",
          },
        },
        "success": Object {
          "style": Object {
            "bg": "green50",
          },
        },
        "warning": Object {
          "style": Object {
            "bg": "orange50",
          },
        },
      },
    },
    "InputCore": Object {
      "props": Object {
        "pb": 1,
      },
    },
    "InputField": Object {
      "clearable": Object {
        "style": Object {
          "pr": "none",
        },
      },
      "iconAfter": Object {
        "style": Object {
          "pr": "none",
        },
      },
      "iconBefore": Object {
        "style": Object {
          "pl": "none",
        },
      },
      "size": Object {
        "l": Object {
          "style": Object {
            "px": 16,
          },
        },
        "m": Object {
          "style": Object {
            "px": 16,
          },
        },
        "s": Object {
          "style": Object {
            "px": 12,
          },
        },
      },
    },
    "Postfix": Object {
      "props": Object {
        "ml": 4,
      },
    },
    "Prefix": Object {
      "props": Object {
        "mr": 4,
      },
    },
  },
}
`;

exports[`mobile theme text theme snap 1`] = `
Object {
  "scheme": Object {
    "Text": Object {
      "dense": Object {
        "style": Object {
          "lineHeight": "dense",
        },
      },
      "preset": Object {
        "heading": Object {
          "props": Object {
            "as": "h4",
            "dense": true,
            "mb": 22,
            "size": "l",
          },
        },
        "heading-large": Object {
          "props": Object {
            "as": "h3",
            "bold": true,
            "dense": true,
            "mb": 22,
            "size": "l",
          },
        },
        "heading-small": Object {
          "props": Object {
            "as": "h5",
            "mb": 16,
            "size": "m",
          },
        },
        "title": Object {
          "props": Object {
            "as": "h1",
            "mb": 36,
            "size": "xxl",
          },
        },
        "title-small": Object {
          "props": Object {
            "as": "h2",
            "bold": true,
            "dense": true,
            "mb": 26,
            "size": "xl",
          },
        },
      },
      "size": Object {
        "l": Object {
          "style": Object {
            "fontSize": "l",
          },
        },
        "m": Object {
          "style": Object {
            "fontSize": "m",
          },
        },
        "s": Object {
          "style": Object {
            "fontSize": "s",
          },
        },
        "xl": Object {
          "style": Object {
            "fontSize": "xl",
          },
        },
        "xs": Object {
          "style": Object {
            "fontSize": "xs",
          },
        },
        "xxl": Object {
          "style": Object {
            "fontSize": "xxl",
          },
        },
        "xxxl": Object {
          "style": Object {
            "fontSize": "xxxl",
          },
        },
      },
      "style": Object {
        "lineHeight": "normal",
      },
      "variant": Object {
        "error": Object {
          "style": Object {
            "color": "red500",
          },
        },
        "primary": Object {
          "style": Object {
            "color": "blue500",
          },
        },
        "secondary": Object {
          "style": Object {
            "color": "gray52",
          },
        },
        "success": Object {
          "style": Object {
            "color": "green500",
          },
        },
        "warning": Object {
          "style": Object {
            "color": "yellow500",
          },
        },
      },
    },
  },
}
`;
